core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1262,6 +1266,12 @@ private View replace(LoadViewResponse response) {
               .max(Integer::compareTo)
               .orElseGet(metadata::currentVersionId);
 
+      schema =
+          TypeUtil.assignFreshIds(
+              schema,
+              metadata.schema(),
+              new AtomicInteger(metadata.schema().highestFieldId())::incrementAndGet);
```

Comment:
This isn't correct.

Shouldn't it be the max of all field IDs across schemas so that the IDs are not reused?

Reply:
There should also be a test case for this.

Reply:
indeed this should be the max of all field IDs across schemas. I've fixed that and added a few tests